spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/sql/TestSelect.java

```diff
@@ -643,4 +644,39 @@ public void testRequiredNestedFieldInOptionalStructFilter() {
     assertEquals("Should return all expected rows", ImmutableList.of(row(0)), result);
     sql("DROP TABLE IF EXISTS %s", nestedStructTable);
   }
+
+  @TestTemplate
+  public void simpleTypesInFilter() {
+    String tableName = tableName("simple_types_table");
+    sql(
+        "CREATE TABLE IF NOT EXISTS %s (id bigint, boolean boolean, integer integer, long long, "
+            + "float float, double double, string string, date date, timestamp timestamp) USING iceberg",
+        tableName);
```

Comment:
Are we planning to add remaining types in follow-up ?
https://github.com/apache/iceberg/blob/0547af078c9d21b8642a4efcb3cbc85756547f22/open-api/rest-catalog-open-api.yaml#L4317-L4334